### PR TITLE
qemu_qtree: Support json format in generate_params for lusk format

### PR DESCRIPTION
**Pull request by YongxueHong**
The value of key "file" is json format from qmp commmand "query-block"
if the backend image format is luks, so need to pick up the "filename"
as the "image_name".

ID: 1838887
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
Hi, @qingwangrh @luckyh please help me to review, thanks.

File: virttest/qemu_qtree.py [hunk 1]
```diff
@@ -216,8 +217,11 @@ def generate_params(self):
             self.params['image_name'] = os.path.realpath(
                 self.block.get('backing_file'))
         elif self.block.get('file'):
-            self.params['image_name'] = os.path.realpath(
-                self.block.get('file'))
+            block_file = self.block.get('file')
+            image_name = os.path.realpath(block_file)
+            if not os.path.exists(block_file) and block_file.startswith('json:'):
+                image_name = json.loads(block_file[5:])['file']['filename']
```
Comment (luckyh) [inline virttest/qemu_qtree.py:223]: Suggest a wrapper function for the usage (e.g. `parse_json(string)`).

File: virttest/qemu_qtree.py [hunk 2]
```diff
@@ -216,8 +217,11 @@ def generate_params(self):
             self.params['image_name'] = os.path.realpath(
                 self.block.get('backing_file'))
         elif self.block.get('file'):
-            self.params['image_name'] = os.path.realpath(
-                self.block.get('file'))
+            block_file = self.block.get('file')
+            image_name = os.path.realpath(block_file)
+            if not os.path.exists(block_file) and block_file.startswith('json:'):
```
Comment (luckyh) [inline virttest/qemu_qtree.py:222]: Looks only using `block_file.startswith('json:')` is enough.

File: virttest/qemu_qtree.py [hunk 3]
```diff
@@ -216,8 +217,11 @@ def generate_params(self):
             self.params['image_name'] = os.path.realpath(
                 self.block.get('backing_file'))
```
Comment (luckyh) [inline virttest/qemu_qtree.py:218]: Is it needed for getting `backing_file`, too?

File: virttest/qemu_qtree.py (hunk 1 above)
Comment (YongxueHong) [reply virttest/qemu_qtree.py]: Thanks, have updated.

File: virttest/qemu_qtree.py (hunk 2 above)
Comment (YongxueHong) [reply virttest/qemu_qtree.py]: Thanks, have updated.

File: virttest/qemu_qtree.py (hunk 3 above)
Comment (YongxueHong) [reply virttest/qemu_qtree.py]: Thanks, have updated.
@XueqiangWei  could you help me to confirm and review? thanks.

**Comment by XueqiangWei** [comment]
Tested with luks, raw, qcow2, all work well.

 (1/1) Host_RHEL.m8.u2.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.multi_disk.one_disk_repeat.q35: PASS (146.47 s)

 (1/1) Host_RHEL.m8.u2.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.multi_disk.one_disk_repeat.q35: PASS (166.20 s)

(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.multi_disk.one_disk_repeat.q35: PASS (140.10 s)

**Comment by YongxueHong** [comment]
@luckyh  please help me to review, thanks.

**Comment by YongxueHong** [comment]
Teste passed with luks.

```
 (1/4) Host_RHEL.m8.u2.luks.virtio_blk.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.all_types.q35: PASS (1634.88 s)
 (2/4) Host_RHEL.m8.u2.luks.virtio_blk.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.q35: PASS (1490.86 s)
 (3/4) Host_RHEL.m8.u2.luks.virtio_blk.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.parallel.all_types.q35: PASS (999.55 s)
 (4/4) Host_RHEL.m8.u2.luks.virtio_blk.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.parallel.single_type.q35: PASS (818.19 s)
```

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**